aws_quickstart/datadog_integration_api_call_v2.yaml

```diff
@@ -65,163 +65,193 @@ Resources:
-                                    {"Message": "Unexpected event received from CloudFormation"})
-              except Exception as e:  # pylint: disable=W0702
+                  # Call Datadog API and report response back to CloudFormation
+                  response = call_datadog_api(event, method)
+                  cfn_response_send_api_result(event, context, method, response)
+
+              except Exception as e:
                   LOGGER.exception('Failed - exception thrown during processing.')
-                  send_response(event, context, "FAILED", {
-                      "Message": "Exception during processing: {}".format(e)})
-
-
-          def send_response(event, context, response_status, response_data):
-              '''Send a resource manipulation status response to CloudFormation'''
-              response_body = json.dumps({
-                  "Status": response_status,
-                  "Reason": "See the details in CloudWatch Log Stream: " + context.log_stream_name,
-                  "PhysicalResourceId": context.log_stream_name,
-                  "StackId": event['StackId'],
-                  "RequestId": event['RequestId'],
-                  "LogicalResourceId": event['LogicalResourceId'],
-                  "Data": response_data
-              })
-              formatted_response = response_body.encode("utf-8")
-
-              LOGGER.info('ResponseURL: %s', event['ResponseURL'])
-              LOGGER.info('ResponseBody: %s', response_body)
-
-              opener = build_opener(HTTPHandler)
-              request = Request(event['ResponseURL'], data=formatted_response)
-              request.add_header('Content-Type', 'application/json; charset=utf-8')
-              request.add_header('Content-Length', len(formatted_response))
-              request.get_method = lambda: 'PUT'
-              response = opener.open(request)
-              LOGGER.info("Status code: %s", response.getcode())
-              LOGGER.info("Status message: %s", response.msg)
+                  msg = {"Message": "Exception during processing: {}".format(e)}
+                  reason = json.dumps(msg)
+                  cfnresponse.send(
+                      event,
+                      context,
+                      "FAILED",
+                      responseData=msg,
+                      reason=reason,
+                  )
+
+
+          def cfn_response_send_api_result(event, context, method, response):
+              reason = None
+              json_response = ""
+              code = response.getcode()
+              data = response.read()
+              if data:
+                  json_response = json.loads(data)
+              if code == 200 or code == 204:
+                  LOGGER.info("Success - Datadog API call was successful.")
+                  response_status = "SUCCESS"
+                  msg = {"Message": "Datadog AWS Integration {} API request was successful.".format(method)}
```

Comment:
I'd name this variable something else. Not sure what's best. Maybe response_payload? Or something else that isn't msg since it can hold more data than just the message.